### PR TITLE
Ajax Loading UI, addresses #91

### PR DESCRIPTION
**Pull request by 22a**
### Don't merge this yet, there is a bug that I will outline below.
I'm just making this pr so that I can add images of the bug.

So there's a few things going on here.

**UI changes:**
* added a beating heart loading icon that takes the place of the form after it has been submitted.
* made it so that the previous results get hidden when the form is submitted a second time (and subsequently updated and show again when we get a response)
* changed the look of the file input (and added some js to preserve the filename display)

These make the user interaction with "slow" (2ish seconds) requests look like this:
The colours are a little off here because gif but you get the picture.

**Code style changes:**
* changed a few of the html id's and javascript variables so that they're more uniform
* started using [Element.classList](https://developer.mozilla.org/en/docs/Web/API/Element/classList) instead of manually setting the display style property with strings

## Bug:
The loading icon blinks in and out when the request responds quickly (which is the case in every circumstance apart from when we upload a valid pml file with ddi's that haven't been cached yet). It looks like this:

The desired solution would be for this blinking to not happen, either through some sort of debouncing or some fading between `display: none` and `display: block`.

### CODE REVIEW

File: panacea/web/static/js/app.js [hunk 1]
```diff
@@ -15,35 +17,61 @@ async function submitFile() {
...
+const restoreFileForm = () => {
+  formElement.classList.remove('hidden');
+  loadingElement.classList.add('hidden');
+}
+
+// to make the file input pretty take the filename from the form
+// and place it in a disabled input box right beside it :art:
+const filenameDisplayElement = document.getElementById('filename-display');
+const fileInputElement = document.getElementById('file-input');
+fileInputElement.onchange = function () {
```
Comment (houli) [inline panacea/web/static/js/app.js]:
```javascript
fileInputElement.addEventListener('change', function(e) {
  filenameDisplayElement.value = this.files[0].name;
});
```

Comment (22a) [reply panacea/web/static/js/app.js]: Much cleaner, thanks @houli 🥇

**Review by c-brenn — COMMENTED** [review]
looks good so far. nice work. any luck with the fading?

**Comment by 22a** [comment]
I been tinkering around, got caught up with the normalise stuff in phoenix.css using the `hidden` classname with `!important`, went in circles there for a while, so any solution will need to not use the `hidden` classname.

I've tried a few things like animation keyframes to fade out then set `height: 0` and a few others but short of writing some js that adds a fading class then removed the element from the dom (adds another class with `display: none` when the animation ends) I can't think of a way to do this, and even then, I don't know what logic we would need to stop such an animation

I think I may be overcomplicating this in my head, the issue is that there are only a handful of milliseconds where the form disappears, the loading icon shows up, and the form reappears when out requests are speedy.

The idea for fading that I had in my head was that the form starts fading out when we send the request, and the loading icon starts fading in at the same time, the issue here is that these two elements would either have to occupy the same space (ergo fade in and out on top of each other) or sit on top of each other vertically, both of which sound janky to me

### What do we want to happen here?

**Review by c-brenn — APPROVED** [review]
🥇 you're a winner

**Comment by c-brenn** [comment]
@22a this should prob go in the change log actually seeing as it's user facing

**Pull request merged**